Improve encoder responsiveness by having it trigger an interrupt

kilrah opened the pull request:
Still misses a click once in a while but nowhere near as often. Was particularly bad on the new faces/memory one

amalmin commented:
Cool.  This helps a lot.  I had been adding doEnc() calls to various places at times during debugging but could never really get it to work well.  It feels like there are switch bouncing problems still though.

kilrah commented:
Yes, in my audio code I've moved the encoder into a class and added debouncing to the encoder pins and it definitely helps. 
I intend to try and refactor the whole original code into classes with each clock/game being an independent "page" to help separate things so it's easier to add stuff including "not-clocks" and things like my audio screens that need direct DAC/controls access
